include/qphix/clover_product.h

```diff
@@ -128,17 +127,20 @@ struct InnerCloverProduct<
                 cplx_mul_acc(spinor_out[c_out][four_s_out][re][xi],
                              spinor_out[c_out][four_s_out][im][xi],
                              diag_in[sc_in][veclen_idx],
-                             FT{0},
+                             QPhiX::rep<FT, double>(0.0),
                              spinor_in[c_in][four_s_in][re][xi],
                              spinor_in[c_in][four_s_in][im][xi]);
               } else if (sc_out < sc_in) {
                 auto const idx15 = sc_in * (sc_in - 1) / 2 + sc_out;
-                cplx_mul_acc(spinor_out[c_out][four_s_out][re][xi],
-                             spinor_out[c_out][four_s_out][im][xi],
-                             off_diag_in[idx15][re][veclen_idx],
-                             -off_diag_in[idx15][im][veclen_idx],
-                             spinor_in[c_in][four_s_in][re][xi],
-                             spinor_in[c_in][four_s_in][im][xi]);
+                cplx_mul_acc(
+                    spinor_out[c_out][four_s_out][re][xi],
+                    spinor_out[c_out][four_s_out][im][xi],
+                    off_diag_in[idx15][re][veclen_idx],
+                    // aww hell, maybe one should just add negation to QPhiX::half ?
+                    QPhiX::rep<FT, double>(
+                        -QPhiX::rep<double, FT>(off_diag_in[idx15][im][veclen_idx])),
```

Comment:
It would probably be sufficient to use `rep<typename ArithType<FT>::Type, FT>` since `ArithType<FT>::Type` becomes `float` for `FT = half`. But this likely is not a problem to just negate with `double` instead of `float`.